app/src/main/java/com/github/se/wanderpals/service/MapManager.kt

```diff
@@ -0,0 +1,186 @@
+package com.github.se.wanderpals.service
+
+import android.Manifest
+import android.content.Context
+import android.content.pm.PackageManager
+import android.util.Log
+import androidx.activity.result.ActivityResultLauncher
+import com.github.se.wanderpals.BuildConfig
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.location.Priority
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.tasks.CancellationTokenSource
+import com.google.android.gms.tasks.Task
+import com.google.android.libraries.places.api.Places
+import com.google.android.libraries.places.api.model.AutocompletePrediction
+import com.google.android.libraries.places.api.model.AutocompleteSessionToken
+import com.google.android.libraries.places.api.model.Place
+import com.google.android.libraries.places.api.model.RectangularBounds
+import com.google.android.libraries.places.api.net.FetchPlaceRequest
+import com.google.android.libraries.places.api.net.FetchPlaceResponse
+import com.google.android.libraries.places.api.net.FindAutocompletePredictionsRequest
+import com.google.android.libraries.places.api.net.FindAutocompletePredictionsResponse
+import com.google.android.libraries.places.api.net.PlacesClient
+import kotlin.coroutines.suspendCoroutine
+import kotlinx.coroutines.tasks.await
+
+/**
+ * Class to manage the map and location services.
+ *
+ * @param context The context of the activity.
+ */
+class MapManager(private val context: Context) {
+
+  private val placeFields =
+      listOf(
+          Place.Field.LAT_LNG,
+          Place.Field.NAME,
+          Place.Field.ADDRESS,
+          Place.Field.BUSINESS_STATUS,
+          Place.Field.PHONE_NUMBER,
+          Place.Field.RATING,
+          Place.Field.USER_RATINGS_TOTAL,
+          Place.Field.WEBSITE_URI,
+          Place.Field.CURRENT_OPENING_HOURS,
+          Place.Field.ICON_URL)
+
+  private lateinit var fusedLocationClient: FusedLocationProviderClient
+  private lateinit var placesClient: PlacesClient
+
+  private lateinit var locationPermissionRequest: ActivityResultLauncher<Array<String>>
+
+  private var startingLocation: LatLng = LatLng(46.519653, 6.632273)
```

Comment:
Is It the intended behavior to start the maps position at this position?

Reply:
I have taken the liberty of starting it at Lausanne since it needs to start somewhere, in later versions we can asks users where they are going and set it there to be more practical